tests/test_processing_stack.py

```diff
@@ -0,0 +1,117 @@
+
+ENVIRONMENT_TAG_KEY = "EnvironmentType"
+ENVIRONMENT_TAG_VAL = os.environ.get("DEPLOY_ENV", "dev")
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def detect_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+    logger.info("Detected Compute Environments: %s", compute_environment_detection_response)
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+            break
+
+    return datalake_compute_environment
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
+    """
+    Trigger immediate launch of computing resources. This will significantly speed up Batch jobs
+    execution.
+    """
+    if (
+        int(datalake_compute_environment["computeResources"]["minvCpus"]) < 1
+        or int(datalake_compute_environment["computeResources"]["desiredvCpus"]) < 1
+    ):
+        batch_client.update_compute_environment(
+            computeEnvironment=datalake_compute_environment["computeEnvironmentName"],
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def detect_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+    logger.info("Detected State Machines: %s", state_machines_detection_response)
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {}
+        for tag in tags_detection_response["tags"]:
+            state_machine_tags[tag["key"]] = tag["value"]
+
+        if (
+            state_machine_tags[ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and state_machine_tags.get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_state_machine = state_machine
+            logger.info("Datalake State Machine: %s", datalake_state_machine)
+            break
+
+    return datalake_state_machine
+
+
+@mark.timeout(1200)
+@mark.infrastructure
+def test_should_successfully_run_dataset_version_creation_process(
+    batch_client, stepfunctions_client
+):
+
+    state_machine_input = """{
+        "dataset_id":"1xyz",
+        "version_id":"2xyz",
+        "type":"RASTER",
+        "metadata_url":"s3://linz-geospatial-data-lake-test-data/tasman_rural_2003_1m/collection.json"
+    }"""
```

Comment:
I would recommend using `dumps({"dataset_id": …})` to create a string out of a `dict` rather than hardcoding a string. That makes it easier to modify, and we won't have to worry about getting the formatting exactly right.

Also, it is not good to rely on any data maintained outside of the test. We should use `utils.Dataset` and [`S3Object`](https://github.com/linz/geospatial-data-lake/pull/179/commits/31a065e6603fe1714a92b2ec4a140cc40cdbe0b0) to create and tear down the information we need.

Reply:
JSON input done in b7b1d46619d9363b0ab138d485a81881332adaae